dlme_airflow/services/harvest_dag_generator.py

```diff
@@ -40,52 +44,67 @@ def default_dag_args():
     }
 
 
-def create_dag(provider, default_args) -> DAG:
+def create_dags(provider) -> list[DAG]:
+    """Returns a list of DAGs for a given provider."""
+    dags = []
+    if provider.catalog.metadata.get("separate_dags"):
+        # create a DAG for each collection, which can be useful if you want to
+        # schedule them separately
+        for collection in provider.collections:
+            dags.append(assemble_dag(collection))
+    else:
+        # create a single DAG for the provider and all its collections which
+        # will all run in parallel
+        dags.append(assemble_dag(provider))
+    return dags
+
+
+def assemble_dag(source: Union[Provider, Collection]):
+    """Returns a DAG for either a Provider or a Collection."""
+    default_args = default_dag_args()
     default_schedule = os.getenv("DEFAULT_DAG_SCHEDULE", "@daily")
 
-    dag = DAG(
-        provider.name,
-        default_args=default_args,
-        schedule_interval=provider.catalog.metadata.get("schedule", default_schedule),
-        start_date=datetime(2022, 9, 6),
-    )
+    schedule = source.catalog.metadata.get("schedule", default_schedule)
+    start_date = datetime(2022, 9, 6)
+    dag_id = source.label()
 
-    with dag:
-        # A dummy operator is required as a transition point between task groups
+    with DAG(
+        dag_id,
+        default_args=default_args,
+        schedule_interval=schedule,
+        start_date=start_date,
+    ) as dag:
         harvest_begin = DummyOperator(
-            task_id="harvest_begin", trigger_rule="none_failed", dag=dag
+            task_id="harvest_begin", trigger_rule="none_failed"
         )
         harvest_complete = DummyOperator(
-            task_id="harvest_complete", trigger_rule="none_failed", dag=dag
+            task_id="harvest_complete", trigger_rule="none_failed"
         )
-
-        # TODO
-        # post_harvest_begin = DummyOperator(task_id='post_harvest_begin', trigger_rule='none_failed', dag=dag)
-        # post_harvest = build_post_havest_taskgroup(provider, dag)
-        # post_harvest_complete = DummyOperator(task_id='post_harvest_complete', trigger_rule='none_failed', dag=dag)
-
-        # TODO: A dummy operator is required as a transition point between task groups
-        # post_harvest_complete = DummyOperator(task_id='post_harvest_complete', trigger_rule='none_failed', dag=dag)
```

Comment:
Hopefully it was ok to remove these comments, I wasn't sure if they were still relevant?